src/Frontend/TableHelper/Animation.js

```diff
@@ -24,227 +19,258 @@ export const stopStatus = { // 暫停時的狀態
+            this.className1 = StartEndBombWeight(WhichComponentSame(id), componentKind.startSearch, componentKind.endSearch, componentKind.bombSearch, componentKind.weightSearch)
+            this.className2 = StartEndBombWeight(WhichComponentSame(id), componentKind.startSearchBomb, componentKind.endSearchBomb, componentKind.bombSearch, componentKind.weightSearchBomb)
         }
-        count += 1
-    }, speed)
+        return this.className = (position.bomb === false) ? this.className1 : this.className2
+    }
+
+    runAnimation(container){
+        this.getItems(container)
+
+        this.id = this.items[0]
+        if(this.id === undefined || container.length === 0){
+            return false
+        }
+        
+        this.getClassName(this.id)
+        setTable(this.id, this.className)
+        return true
+    }
+    
 }
 
-export function SearchAnimation(bomb, path, pathDirection, speed, pathFunction, sysStatusFunction, updateFunction) {
-    var count = stopStatus.search[0]
+class searchAnimation_bomb extends searchAnimation_pos{
+    getClassName(id){
+        if (WhichComponentSame(id) > 3){
+            this.className = componentKind.search
+        } else {
+            this.className = StartEndBombWeight(WhichComponentSame(id), componentKind.startSearch, componentKind.endSearch, componentKind.bombSearch, componentKind.weightSearch)
+        }
+    }
 
-    const searchAnimation = setInterval(() => {
-        if (count === bomb.length) {
-            stopStatus.search = [bomb.length, 0]
-            pathFunction(path, speed, pathDirection, sysStatusFunction, updateFunction)
-            clearInterval(searchAnimation)
+    runAnimation(container){
+        this.getItems(container)
 
-        }else{
-            for(var i = (count === stopStatus.search[0]) * stopStatus.search[1];i < bomb[count].length;i++){
+        this.id = this.items[0]
+        if(this.id === undefined || container.length === 0){
+            return false
+        }
+        
+        this.getClassName(this.id)
+        setTable(this.id, this.className)
+        return true
+    }
+    
+}
 
-                if(stopStatus.animationStatus === false){
-                    stopStatus.search = [count, i]
-                    sysStatusFunction()
-                    clearInterval(searchAnimation)
-                    return
+class searchAnimation_path extends searchAnimation_pos{
+    getItems(path, pathDirection){
+        this.items =  (path.length > 0) ? [path.shift(), pathDirection.shift()] : [undefined]
+    }
 
-                }else{
-                    if (WhichComponentSame(bomb[count][i]) > 3) {
-                        setTable(bomb[count][i], componentKind.search)
-                    }else{
-                        setTable(bomb[count][i], StartEndBombWeight(WhichComponentSame(bomb[count][i]), componentKind.startSearch, componentKind.endSearch, componentKind.bombSearch, componentKind.weightSearch))
-                    }
+    getClassName(id, d){
```

Comment:
```suggestion
    getClassName(id, d){
        this.className1 = componentKind.path
        this.className2 = WhichComponentSame(id) > 3 ? direction(d) : StartEndBombWeight(WhichComponentSame(id), direction(d), direction(d), componentKind.bombPath, componentKind.weightPath)
    }
```
Be able to use ternary condition operator